utils/awsutils/session.go

```diff
@@ -110,5 +111,13 @@ func NewSessionConfig(destinationConfig map[string]interface{}, timeout time.Dur
 	}
 	sessionConfig.Timeout = timeout
 	sessionConfig.Service = serviceName
+	if sessionConfig.IAMRoleARN != "" {
+		/**
+		In order prevent confused deputy problem, we are using
+		workspace token as external ID.
+		Ref: https://docs.aws.amazon.com/IAM/latest/UserGuide/confused-deputy.html
+		*/
+		sessionConfig.ExternalID = config.GetWorkspaceToken()
```

Comment:
`GetWorkspaceToken` would return the same value for all the customers on free deployment and for the new multi-tenant deployments. Is this okay to have it same for every customer?

Reply:
No, that'll be an issue. How to get each customer's workspace token?

Reply:
We decided to propagate workspace ID to destination handlers and use it as external id. @saikumarrs